source/h2_stream.c

```diff
@@ -749,13 +747,24 @@ struct aws_h2err aws_h2_stream_on_decoder_headers_i(
 
     } else {
         /* Client */
-        if (name_enum == AWS_HTTP_HEADER_STATUS) {
-            uint64_t status_code;
-            int err = aws_byte_cursor_utf8_parse_u64(header->value, &status_code);
-            AWS_ASSERT(!err && "Invalid :status value. Decoder should have already validated this");
-            (void)err;
-
-            stream->base.client_data->response_status = (int)status_code;
+        switch (name_enum) {
+            case AWS_HTTP_HEADER_STATUS: {
+                uint64_t status_code = 0;
+                int err = aws_byte_cursor_utf8_parse_u64(header->value, &status_code);
+                AWS_ASSERT(!err && "Invalid :status value. Decoder should have already validated this");
+                (void)err;
+
+                stream->base.client_data->response_status = (int)status_code;
+            } break;
+            case AWS_HTTP_HEADER_CONTENT_LENGTH: {
+                if (aws_byte_cursor_utf8_parse_u64(header->value, &stream->thread_data.content_length)) {
+                    AWS_H2_STREAM_LOG(ERROR, stream, "Invalid content-length value");
+                    goto malformed;
+                }
+                stream->thread_data.content_length_received = true;
```

Comment:
check to ensure that we're not getting duplicate content-length headers. Request smuggling!